### DISCUSSION TITLE
외부 개발사 자체 결제 내역 전송 API 호출시 403 오류 문의

### DISCUSSION THREAD

**Discussion by pitzcarraldo** [commerce-api-naver/commerce-solutions#40] [묻고 답하기] [answered]
https://apicenter.commerce.naver.com/ko/basic/commerce-api#tag/%EC%99%B8%EB%B6%80-%EA%B0%9C%EB%B0%9C%EC%82%AC-%EC%9E%90%EC%B2%B4-%EA%B2%B0%EC%A0%9C/operation/sendTransactionExternal.merchant

상기 API  연동 과정에서 안내된 Path,Payload로 호출하였으나 다음과 같은 응답을 받고 있습니다.
동일 토큰, 동일 솔루션에서 다른 API 호출시에는 권한 오류 없이 정상 응답을 받고 있으나 해당 API에서만 403 오류가 나고 있어서 문의 드립니다.
```js
{
  code: "GW.AUTHN",
  message: "요청을 보낼 권한이 없습니다.",
  timestamp: "2024-10-25T16:03:27.101+09:00",
  traceId: "cr3-165333-678pfx^1729065282417^85844009"
}
```

솔루션 설정 페이지에서도 해당 API의 접근 권한은 필수 선택으로 되어있어서 컨트롤이 가능한 부분이 아닌 것 같습니다. (항상 enable)
<img width="837" alt="Screenshot 2024-10-25 at 4 07 15 PM" src="https://github.com/user-attachments/assets/7dea3668-3642-45e4-a535-0b22f96ec4a2">

호출 path: https://api.commerce.naver.com/external/v1/commerce-solutions/external-transactions
호출 method: POST
사용한 payload: 
```json
{
  "transactionType": "PAYMENT",
  "solutionId": "SOL_***",
  "paymentId": "NAVER-COMMERCE-TEST_1729840226040",
  "accountUid": "ncp_***",
  "totalAmount": 1000,
  "paymentConfirmDate": "2024-10-25T07:10:26.040Z",
  "paymentModel": "MONTHLY"
}
```

**Comment by commerce-api-naver** [comment, marked as answer]
안녕하세요. 커머스API 이민석입니다.

> [!IMPORTANT]
> 게시해주신 내용은 커머스솔루션 개발사 연동 관련 문의로 판단되어 관리자가 게시 공간을 이동하였습니다.
> - 최초 게시된 공간: 커머스API 내 스토어 애플리케이션 기술지원 공간 (`/commerce-api`)
> - 이동된 공간: 커머스솔루션 개발사 기술지원 공간 (`/commerce-solutions`)

호출을 시도하신 대상 API는 솔루션 개발사 자기 자신(type: SELF)에 대한 인증 토큰을 이용하여 호출이 가능합니다.
알려주신 호출건(Trace ID: cr3-165333-678pfx^1729065282417^85844009)에 사용하신 인증 토큰은 솔루션 구독자에 대한 리소스 권한을 갖는(type: SELLER) 인증 토큰이기에 문의주신 오류 현상이 발생하였습니다.

[커머스솔루션] API 그룹 내 대부분의 API는 솔루션 개발사의 자기 자신에 대한 리소스 권한을 갖는 인증 토큰으로 호출이 가능합니다.
인증 토큰의 발급 유형(`type` 파라미터)를 변경하시어 발급받으신 인증 토큰으로 재호출을 시도해보시기 바랍니다.
